core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/source/scan/csv/CSVScanSourceOpDesc.scala

```diff
@@ -1,57 +1,65 @@
 package edu.uci.ics.texera.workflow.operators.source.scan.csv
 
-import com.fasterxml.jackson.annotation.{JsonIgnore, JsonProperty, JsonPropertyDescription}
-import com.google.common.io.Files
+import com.fasterxml.jackson.annotation.{JsonProperty, JsonPropertyDescription}
+import com.github.tototoshi.csv.{CSVReader, DefaultCSVFormat}
 import com.kjetland.jackson.jsonSchema.annotations.JsonSchemaTitle
 import edu.uci.ics.amber.engine.common.Constants
+import edu.uci.ics.amber.engine.operators.OpExecConfig
 import edu.uci.ics.texera.workflow.common.tuple.schema.{Attribute, AttributeType, Schema}
 import edu.uci.ics.texera.workflow.common.tuple.schema.AttributeTypeUtils.inferSchemaFromRows
 import edu.uci.ics.texera.workflow.operators.source.scan.ScanSourceOpDesc
 import org.codehaus.jackson.map.annotate.JsonDeserialize
 
-import java.io.{BufferedReader, File, FileReader, IOException}
-import java.nio.charset.Charset
-import scala.collection.JavaConverters._
+import java.io.IOException
+import scala.jdk.CollectionConverters.asJavaIterableConverter
 
 class CSVScanSourceOpDesc extends ScanSourceOpDesc {
 
-  @JsonIgnore
-  var headerLine: Option[String] = None
-
   @JsonProperty(defaultValue = ",")
   @JsonSchemaTitle("Delimiter")
   @JsonPropertyDescription("delimiter to separate each line into fields")
   @JsonDeserialize(contentAs = classOf[java.lang.String])
-  var delimiter: Option[String] = None
+  var customDelimiter: Option[String] = None
 
   @JsonProperty(defaultValue = "true")
   @JsonSchemaTitle("Header")
   @JsonPropertyDescription("whether the CSV file contains a header line")
   var hasHeader: Boolean = true
 
+  @JsonProperty(defaultValue = "false")
+  @JsonSchemaTitle("Multiline data?")
+  @JsonPropertyDescription("whether the CSV file contains data that span multi-lines")
```

Comment:
the library can be useful when the (1) the field contains the delimiter, (2) when the fields are wrapped with double quotes, (3) with a custom delimiter other than ',' (4) and for multilines. So the description can be misleading because it is too specific to one case only.

Reply:
Do you have a better name for it?

CSV would add quotation marks only when it has ambiguities, for example, if the field contains a delimiter, or the field is spanning multiple lines. So basically 1) 2) and 4) are the same thing: to remove ambiguity.

Our custom CSV (ParallelCSV) is already supporting custom delimiter. The biggest noticeable difference between two CSV Exec implementations is whether it supports multiline data. We want to hide the technical details from users, it would be great if we can think of a more friendly name.

Reply:
I have a simple file that uses '|' as the delimiter and it fails on ParallelCSV. I have another file that has ',' in one of its fields and it fails in ParallelCSV. So I believe it is better to be very clear when to use what.

Reply:
do they fail even when they are quoted?

Reply:
Yes! I can share the files with you